ios/tunnel/untrusted.go

```diff
@@ -0,0 +1,483 @@
+	if err != nil {
+		return err
+	}
+
+	var responseEvent pairingData
+	err = t.controlChannel.readEvent(&responseEvent)
+	if err != nil {
+		return err
+	}
+
+	errRes, err := tlvReader(responseEvent.data).readCoalesced(typeError)
+	if err != nil {
+		return err
+	}
+	if len(errRes) > 0 {
+		log.Debug("send pair verify failed event")
+		err := t.controlChannel.writeEvent(pairVerifyFailed{})
+		if err != nil {
+			return err
+		}
+		return tlvError(errRes[0])
+	}
+
+	err = t.setupCiphers(sharedSecret)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+type tunnelListener struct {
+	PrivateKey      *rsa.PrivateKey
+	DevicePublicKey interface{}
+	TunnelPort      uint64
+}
+
+type tunnelParameters struct {
+	ServerAddress    string
+	ServerRSDPort    uint64
+	ClientParameters struct {
+		Address string
+		Netmask string
+		Mtu     uint64
+	}
+}
+
+func (t *tunnelService) setupSessionKey() ([]byte, error) {
+	devicePublicKey, deviceSalt, err := t.readDeviceKey()
+	if err != nil {
+		return nil, fmt.Errorf("setupSessionKey: failed to read device public key and salt value: %w", err)
+	}
+
+	srp, err := newSrpInfo(deviceSalt, devicePublicKey)
+	if err != nil {
+		return nil, fmt.Errorf("setupSessionKey: failed to setup SRP: %w", err)
+	}
+
+	proofTlv := newTlvBuffer()
+	proofTlv.writeByte(typeState, pairStateVerifyRequest)
+	proofTlv.writeData(typePublicKey, srp.ClientPublic)
+	proofTlv.writeData(typeProof, srp.ClientProof)
+
+	err = t.controlChannel.writeEvent(&pairingData{
+		data: proofTlv.bytes(),
+		kind: "setupManualPairing",
+	})
+	if err != nil {
+		return nil, fmt.Errorf("setupSessionKey: failed to send SRP proof: %w", err)
+	}
+
+	var proofPairingData pairingData
+	err = t.controlChannel.readEvent(&proofPairingData)
+	if err != nil {
+		return nil, fmt.Errorf("setupSessionKey: failed to read device SRP proof: %w", err)
+	}
+
+	serverProof, err := tlvReader(proofPairingData.data).readCoalesced(typeProof)
+	if err != nil {
+		return nil, fmt.Errorf("setupSessionKey: failed to parse device proof: %w", err)
+	}
+	verified := srp.verifyServerProof(serverProof)
+	if !verified {
+		return nil, fmt.Errorf("setupSessionKey: could not verify server proof")
+	}
+	return srp.SessionKey, nil
+}
+
+func (t *tunnelService) exchangeDeviceInfo(sessionKey []byte) error {
+	hkdfPairSetup := hkdf.New(sha512.New, sessionKey, []byte("Pair-Setup-Controller-Sign-Salt"), []byte("Pair-Setup-Controller-Sign-Info"))
+	buf := bytes.NewBuffer(nil)
+	io.CopyN(buf, hkdfPairSetup, 32)
+	buf.WriteString(t.pairRecords.selfId.Identifier)
+	buf.Write(t.pairRecords.selfId.publicKey())
+
+	signature := ed25519.Sign(t.pairRecords.selfId.privateKey(), buf.Bytes())
+
+	deviceInfo, err := opack.Encode(map[string]interface{}{
+		"accountID":                   t.pairRecords.selfId.Identifier,
+		"altIRK":                      []byte{0x5e, 0xca, 0x81, 0x91, 0x92, 0x02, 0x82, 0x00, 0x11, 0x22, 0x33, 0x44, 0xbb, 0xf2, 0x4a, 0xc8},
+		"btAddr":                      "FF:DD:99:66:BB:AA",
```

Comment:
can you put these things in constants maybe or somehow explain why they are hardcoded?